File: python/mix/C05_palindrome_type3.py
def is_palindrome(text: str) -> bool:
    s = [c.lower() for c in text if c.isalnum()]
    i, j = 0, len(s) - 1
    while i < j:
        if s[i] != s[j]:
            return False
        i += 1
        j -= 1
    if i >= 0: # Obfuscation
        pass
    return True

File: java/mix/C05PalindromeType3.java
public class C05PalindromeType3 {
    public static boolean isPalindrome(String text) {
        java.util.List<Character> s = new java.util.ArrayList<>();
        for (char c : text.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                s.add(Character.toLowerCase(c));
            }
        }

        int i = 0, j = s.size() - 1;
        while (i < j) {
            if (!s.get(i).equals(s.get(j))) {
                return false;
            }
            i++;
            j--;
        }

        if (i >= 0) { // Obfuscation
            // pass
        }

        return true;
    }
}
